app/utils/extract_content.py

```
@@ -0,0 +1,228 @@
+        for img_idx, image in enumerate(images):
+            try:
+                xref = image[0]
+                base_image = doc.extract_image(xref)
+                image_bytes = base_image["image"]
+
+                pil_image = Image.open(io.BytesIO(image_bytes)).convert("RGB")
+
+                # In some cases get_image_bbox can fail; guard it
+                try:
+                    img_rect = page.get_image_bbox(image)
+                except Exception:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)